### ISSUE TITLE
A DSU confirms to a mobile device that it has received the data

### ISSUE THREAD

**Issue by zhtaoxiang** [zhehaowang/ndnfit#3]
Now Tcp Bulk Insert Repo Insertion Protocol is used to insert data into repo.

When a mobile device sends a request to a DSU to confirm that the DSU has received the data, the DSU does not have a good way to make sure that it has received the data (send data interest to the repo to check if the data is there will not be a choice, because this interest may also be routed to the Android device again. That will make the confirmation always succeeds).

The solution is to use standard protocol to insert data and use standard way to check insertion status.

**Comment by zhehaowang** [comment]
> The solution is to use standard protocol to insert data and use standard way to check insertion status

Yes this is one way; in which case the DSU client will need to get the insertion process ID from repo-ng, and forward that to the mobile client. The mobile client will need to use this process ID to query whether an insertion's done.

Another way is for DSU to use out-of-band mechanism (for example, TCP), which you may already have now.

This also has something to do with what we discussed today: essentially identifying DSU client's interest in a way so that it only gets answered by the local repo, or never gets answered by the local repo.

**Comment by zhtaoxiang** [comment]
For now, I am using the out-of-band mechanism, I just want to keep this issue open to see if we have better solutions.
